### DISCUSSION TITLE
Deregister a host

### DISCUSSION THREAD

**Discussion by dwwilde** [smallstep/certificates#770] [Q&A] [answered]
I am running the open source step-ca server using a docker container.   I have everything working correctly and have been very impresses with the project.   Naturally, I have been adding many hosts in my testing period using a command like:
step ssh certificate server.domain.edu ssh_host_ecdsa_key.pub --host --sign --principal server.domain.edu
step ssh config --host --set Certificate=ssh_host_ecdsa_key-cert.pub --set Key=ssh_host_ecdsa_key
to setup my sshd and such.
After this step a "step ssh hosts" lists the new server.
My question is how do I remove this server from the step-ca database if I wanted to stop using step-ca to authenticate logins for it?  I know that I can remove/revert the settings on the host back to the original settings and the host will be fine.  But on any client the "step ssh hosts" command will still show that entry so the match exec will succeed and the step ssh proxycommand will try to run.
Thanks,
Dave

**Comment by maraino** [comment, marked as answer]
Hi @dwwilde, unfortunately, that functionality is still not available in the open-source version of step-ca. Currently is only available in [Smallstep SSH](https://smallstep.com/sso-ssh/).

However, there are manual ways to actually do it, they depend on the database backend that you're using. Basically delete the proper entries in the tables `ssh_hosts` and `ssh_host_principals`, which might be easy in MySQL but for Badger, you will need to create a custom program to do it.
